Jellyfin.Server/Migrations/Routines/PopulateImageSortOrder.cs

```diff
@@ -0,0 +1,239 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Jellyfin.Database.Implementations;
+using Jellyfin.Database.Implementations.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Server.Migrations.Routines;
+
+/// <summary>
+/// Populate SortOrder for existing images after the AddSortOrderToBaseItemImageInfo migration.
+/// </summary>
+#pragma warning disable CS0618 // Type or member is obsolete
+[JellyfinMigration("2025-10-23T02:30:00", nameof(PopulateImageSortOrder), "A7B3F8E9-4C2D-4E1A-9B5F-6D8E3F2A1C9B")]
+internal class PopulateImageSortOrder : IMigrationRoutine
+#pragma warning restore CS0618 // Type or member is obsolete
+{
+    private const int UnknownImagePriority = 999;
+
+    private readonly IDbContextFactory<JellyfinDbContext> _dbProvider;
+    private readonly ILogger<PopulateImageSortOrder> _logger;
+
+    public PopulateImageSortOrder(
+        IDbContextFactory<JellyfinDbContext> dbProvider,
+        ILogger<PopulateImageSortOrder> logger)
+    {
+        _dbProvider = dbProvider;
+        _logger = logger;
+    }
+
+    /// <inheritdoc/>
+    public void Perform()
+    {
+        _logger.LogInformation("Starting SortOrder population for existing images");
+
+        using var context = _dbProvider.CreateDbContext();
+
+        // Get all distinct ItemIds that have images
+        var itemIdsWithImages = context.BaseItemImageInfos
+            .Select(i => i.ItemId)
+            .Distinct()
+            .ToList();
+
+        if (itemIdsWithImages.Count == 0)
+        {
+            _logger.LogInformation("No items with images found, skipping SortOrder population");
+            return;
+        }
+
+        _logger.LogInformation("Processing {Count} items with images", itemIdsWithImages.Count);
+
+        // Batch processing configuration
+        const int batchSize = 500;
+        int processedCount = 0;
+        int errorCount = 0;
+
+        // Load all base item paths into memory for fast lookup
+        var itemPathLookup = context.BaseItems
```

Comment:
Loading all BaseItems into memory also pushes a lot of ram which isn't nessesary here.